pkg/operator/ceph/cluster/mon/drain.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mon
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	monPDBName = "rook-ceph-mon-pdb"
+)
+
+func (c *Cluster) reconcileMonPDB() error {
+	monCount := c.spec.Mon.Count
+	if monCount <= 2 {
+		logger.Debug("managePodBudgets is set, but mon-count <= 2. Not creating a disruptionbudget for Mons")
+		return nil
+	}
+
```

Comment:
If the pdb is not enabled, we also need to delete the pdb as well.

Reply:
How about handling deletion of all PDB (mon and OSD) in a separate PR?  (if we handle mon here, we would need to handle osds as well else `ManagePodBudgets=false` would only delete mon PDB)

Reply:
It's fine if that one setting is applied in different controllers for the two different PDBs, I don't see why the same controller has to delete the PDBs. This controller owns the mon pdb and the osd pdb is owned by the disruption controller. In a separate PR is fine though, it's not critical for this PR.